chainercv/links/model/ssd/multibox_coder.py

```diff
@@ -0,0 +1,231 @@
+from __future__ import division
+
+import itertools
+import numpy as np
+
+import chainer
+
+from chainercv import utils
+
+
+class MultiboxCoder(object):
+    """A helper class to encode/decode bounding boxes.
+
+    This class encodes :obj:`(bbox, label)` to :obj:`(mb_loc, mb_label)`
+    and decodes :obj:`(mb_loc, mb_conf)` to `(bbox, label, score)`.
+    These encoding/decoding are used in Single Shot Multibox Detector [#]_.
+
+    .. [#] Wei Liu, Dragomir Anguelov, Dumitru Erhan, Christian Szegedy,
+       Scott Reed, Cheng-Yang Fu, Alexander C. Berg.
+       SSD: Single Shot MultiBox Detector. ECCV 2016.
+
+    Args:
+        grids (iterable of ints): An iterable of integers.
+            Each integer indicates the size of feature map.
+        aspect_ratios (iterable of tuples of ints)`:
+            An iterable of tuples of integers.
+            Each tuple indicates the aspect ratios of default bounding boxes
+            at each feature maps.
+            The length of this iterable should be :obj:`len(grids)`.
+        steps (iterable of floats): The step size for each feature map.
+            The length of this iterable should be :obj:`len(grids)`.
+        sizes (iterable of floats): The base size of default bounding boxes
+            for each feature map.
+            The length of this iterable should be :obj:`len(grids) + 1`.
+        variance (tuple of floats): Two coefficients for encoding/decoding
+            the locations of bounding boxe. The first value is used to
+            encode/decode coordinates of the centers.
+            The second value is used to encode/decode the sizes of
+            bounding boxes.
+    """
+
+    def __init__(self, grids, aspect_ratios, steps, sizes, variance):
+        if not len(aspect_ratios) == len(grids):
+            raise ValueError('The length of aspect_ratios is wrong.')
+        if not len(steps) == len(grids):
+            raise ValueError('The length of steps is wrong.')
+        if not len(sizes) == len(grids) + 1:
+            raise ValueError('The length of sizes is wrong.')
+
+        default_bbox = list()
+
+        for k, grid in enumerate(grids):
+            for v, u in itertools.product(range(grid), repeat=2):
+                cy = (v + 0.5) * steps[k]
+                cx = (u + 0.5) * steps[k]
+
```

Comment:
The order of default bounding boxes is important. Current order is same as the original implementation. I think your suggestion will change the order.